pkg/hook/testdata/github.com/initial/sql.go

```diff
@@ -0,0 +1,275 @@
+		var name string
+		if err := rows.Scan(&name); err != nil {
+			// Check for a scan error.
+			// Query rows will be closed with defer.
+			log.Fatal(err)
+		}
+		names = append(names, name)
+	}
+	// If the database is being written to ensure to check for Close
+	// errors that may be returned from the driver. The query may
+	// encounter an auto-commit error and be forced to rollback changes.
+	rerr := rows.Close()
+	if rerr != nil {
+		log.Fatal(rerr)
+	}
+
+	// Rows.Err will report the last error encountered by Rows.Scan.
+	if err := rows.Err(); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Printf("%s are %d years old", strings.Join(names, ", "), age)
+}
+
+func updateUsersExec(status string, id int, withContext bool) {
+	tx, err := db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if withContext {
+		_, err = tx.ExecContext(ctx, `UPDATE users SET status = ? WHERE id = ?`, "paid", id)
+	} else {
+		_, err = tx.Exec(`UPDATE users SET status = ? WHERE id = ?`, "paid", id)
+	}
+	if err != nil {
+		_ = tx.Rollback()
+		log.Fatal(err)
+	}
+	if err := tx.Commit(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func updateUsersQuery(status string, id int, withContext bool) {
+	tx, err := db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if withContext {
+		_, err = tx.QueryContext(ctx, `UPDATE users SET status = ? WHERE id = ?`, "paid", id)
+	} else {
+		_, err = tx.Query(`UPDATE users SET status = ? WHERE id = ?`, "paid", id)
+	}
+	if err != nil {
+		_ = tx.Rollback()
+		log.Fatal(err)
+	}
+	if err := tx.Commit(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func selectUserDb(id int, withContext bool) {
+	var err error
+	var username string
+	var created time.Time
+	if withContext {
+		err = db.QueryRowContext(ctx, "SELECT username, created_at FROM users WHERE id=?", id).Scan(&username, &created)
+	} else {
+		err = db.QueryRow("SELECT username, created_at FROM users WHERE id=?", id).Scan(&username, &created)
+	}
+	switch {
+	case err == sql.ErrNoRows:
+		log.Printf("no user with id %d\n", id)
+	case err != nil:
+		log.Fatalf("query error: %v\n", err)
+	default:
+		log.Printf("username is %q, account created on %s\n", username, created)
+	}
+}
+
+func selectUserTx(id int, withContext bool) {
+	tx, err := db.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelSerializable})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	var username string
+	var created time.Time
+	if withContext {
+		err = tx.QueryRowContext(ctx, "SELECT username, created_at FROM users WHERE id=?", id).Scan(&username, &created)
+	} else {
+		err = tx.QueryRow("SELECT username, created_at FROM users WHERE id=?", id).Scan(&username, &created)
+	}
+	switch {
+	case err == sql.ErrNoRows:
+		log.Printf("no user with id %d\n", id)
+	case err != nil:
+		log.Fatalf("query error: %v\n", err)
+	default:
+		log.Printf("username is %q, account created on %s\n", username, created)
+	}
+}
```

Comment:
If we add the `Conn` type we should add another test case.

Reply:
Added